configuration/config2/ConfigPath.hpp

```diff
@@ -0,0 +1,86 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+
+#ifndef _CONFIG2_CONFIGPATH_HPP_
+#define _CONFIG2_CONFIGPATH_HPP_
+
+#include <string>
+#include <vector>
+#include <memory>
+
+class ConfigPathItem
+{
+    public:
+
+        ConfigPathItem() : m_isParent(false), m_isCurrent(false), m_isSchemaItem(false), m_isRoot(false) {}
+        ConfigPathItem(const std::string &elemName) : m_isParent(false), m_isCurrent(false), m_isSchemaItem(false), m_isRoot(false), m_elementName(elemName) {}
+
+        void setElementName(const std::string &elemName) { m_elementName = elemName; }
+        const std::string &getElementName() const { return m_elementName; }
+        void setAttributeName(const std::string &attrName) { m_attributeName = attrName; }
+        const std::string &getAttributeName() const { return m_attributeName; }
+        void addAttributeValue(const std::string &attrValue) { m_attributeValues.push_back(attrValue); }
+        bool hasAttributeValues() const { return m_attributeValues.size() > 0; }
+        std::vector<std::string> getAttributeValues() const { return m_attributeValues; }
+        bool isValuePresent(const std::string val, bool returnTrueIfValueListEmpty = true) const;
+        void setIsCurrentElement(bool current) { m_isCurrent = current; }
+        bool isCurrentElement() const { return m_isCurrent; }
+        void setIsParentElement(bool parent) { m_isParent = parent; }
+        bool isParentElement() const { return m_isParent; }
```

Comment:
hasChildren? isParentElement sounds more like isThisTheParentOf(obj potentialchild)

Reply:
got it.